### ISSUE TITLE
Unable to see logs for initialization

### ISSUE THREAD

**Issue by andschdk** [mixpanel/mixpanel-swift#120]
While playing around with the automatic push notifications i noticed that the delegate functions was not swizzled because my `UIApplicationDelegate` was not a `UNUserNotificationCenterDelegate`.

Enabling logging like this results in a crash:
```swift
Mixpanel.mainInstance().loggingEnabled = true
Mixpanel.initialize(token: "token-here")
```

Because of that, one is unable to see all emitted log statements from initialization.

**Comment by yarneo** [comment]
Hey @andschdk, could you please share a crash log? I don't think I fully understand this issue. One delegate cannot be another delegate (or implement it). If you mean that your main UIApplication doesn't implement `UNUserNotificationCenterDelegate` then that's fine. What we do is we swizzle the `UNUserNotificationCenterDelegate` that is received from the `UNUserNotificationCenter.current` as seen here: https://github.com/mixpanel/mixpanel-swift/blob/master/Mixpanel/AutomaticEvents.swift#L171

If you can provide more info on the crash i'll be happy to help!

**Comment by andschdk** [comment]
Hi @yarneo ,
Yes, I mean that my class implementing `UIApplicationDelegate` is not also implementing `UNUserNotificationCenterDelegate`.

I see that you find the "selector" from the new `UNUserNotificationCenter.current`, but it still looks like you are passing the app delegate as argument to the swizzling function: https://github.com/mixpanel/mixpanel-swift/blob/master/Mixpanel/AutomaticEvents.swift#L199
I have opened separate issue for this https://github.com/mixpanel/mixpanel-swift/issues/122

Back to the logging issue..
I just noticed that I do in fact see the logging for the swizzling when setting `loggingEnabled` after `initialize` like so:
```swift
Mixpanel.initialize(token: "token-here")
Mixpanel.mainInstance().loggingEnabled = true
```

My apologise for the confusion.

I think the crash is fair as it was in `Mixpanel`:

```swift
fatalError("You have to call initialize(token:) before calling the main instance, " +
                "or define a new main instance if removing the main one")
```

I'll close this one, thanks.
